qa/integration-tests/src/test/java/io/camunda/zeebe/it/startup/AbstractRootlessImageTest.java

```diff
@@ -11,52 +11,34 @@
 
 import io.camunda.zeebe.client.ZeebeClient;
 import io.camunda.zeebe.client.api.response.ProcessInstanceResult;
+import io.camunda.zeebe.it.smoke.ContainerSmokeTest;
 import io.camunda.zeebe.model.bpmn.Bpmn;
 import io.camunda.zeebe.qa.util.testcontainers.ContainerLogsDumper;
 import io.camunda.zeebe.qa.util.testcontainers.ZeebeTestContainerDefaults;
 import io.zeebe.containers.ZeebeBrokerContainer;
 import io.zeebe.containers.ZeebeGatewayContainer;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 import org.testcontainers.containers.Network;
 import org.testcontainers.junit.jupiter.Container;
-import org.testcontainers.junit.jupiter.Testcontainers;
 
-/**
- * Tests a small deployment of one standalone broker and gateway, running "rootless". While this
- * cannot guarantee OpenShift compatibility, it's the most common compatibility issue.
- *
- * <p>See <a
- * href="https://docs.openshift.com/container-platform/latest/openshift_images/create-images.html">here</a>
- * for more.
- *
- * <p>From their docs: By default, OpenShift Container Platform runs containers using an arbitrarily
- * assigned user ID. This provides additional security against processes escaping the container due
- * to a container engine vulnerability and thereby achieving escalated permissions on the host node.
- * For an image to support running as an arbitrary user, directories and files that are written to
- * by processes in the image must be owned by the root group and be read/writable by that group.
- * Files to be executed must also have "group" execute permissions.
- *
- * <p>You can read more about UIDs/GIDs <a
- * href="https://cloud.redhat.com/blog/a-guide-to-openshift-and-uids">here</a>.
- */
-@Testcontainers
-final class RootlessImageTest {
+/** Tests a small deployment of one standalone broker and gateway, running "rootless". */
+abstract class AbstractRootlessImageTest {
```

Comment:
:upside_down_face: Just a personal opinion: using an abstract class for this feels very heavy handed. I think this could just be a parameterized tests if the single test case just starts the container directly instead of using class fields and the `@Container` annotation.

Reply:
Agreed, abstract test classes are a gateway to the dark side :smile: